department-of-veterans-affairs/va.gov-cms
# DNM: VACMS-15385: Event Email CTA

# Do not merge until the FE work is ready.

## Description
This PR adds the ability to define an Email address for an Event CTA. In order to allow the editor to specify URL or Email, an additional field, 'How to apply', controls the visibility of the URL and Email fields based on its value.

The flow of setting a CTA has changed, such that once a user chooses any 'Call to Action' value other than 'none', the new 'how to apply' field is revealed and requires a user to set either an Email or URL value.

Summary of changes
- Adds a new 'How to apply' field to the Event content type
- Adds a new 'Email" field to the Event content type
- New constraint to prevent setting a mailto: in the URL field
- New constraint to prevent setting the Email to any address outside of va.gov
- Updates existing Event client-side javascript to control visibility for the new fields.
- Updates/add cypress tests

Relates to #15385

## Screenshots

### Before
![Before - Event registration URL](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/eb2e6297-f966-4724-9a78-cf507eb43f6a)

### After
![After - Event registration no CTA](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/b6eb51d9-b216-460f-affe-812eeec7893c)
![After - Event registration Email](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/3f66bd4c-2f3c-4fe5-b867-fe5c5db8ff2f)
![After - Event registration URL](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/4e675b39-a6a1-4273-8ec8-b90f576bebc6)
![After - Event registration How to Sign Up](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/221539/9fbd25af-d399-48b8-a363-36c4d0edc3fd)

## QA steps
[TBD]
What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

### Test existing Event with URL

As a content admin

1. visit https://cms-fwlvssypud9glye0p1gfv4hmamkufgpi.ci.cms.va.gov/node/44574/edit
2. Scroll to the 'Registration' fieldset
   - [x] Validate that 'Call to Action' is set to 'Register'
   - [x] Validate that 'How to Sign Up' is set to 'URL'
   - [x] Validate that 'URL' is set to 'https://dashboard.teletownhall.us/registration/CP4BJYA9XFM2'

### Test field conditional display

As a content admin

3. Visit https://cms-fwlvssypud9glye0p1gfv4hmamkufgpi.ci.cms.va.gov/node/add/event
4. Select the 'Include registration information' checkbox
   - [x] Validate the 'Call to action' field is set to "- None - "
5. Select 'Register'
   - [x] Validate the 'How to apply' field appears.
6. Select 'URL' in the 'How to apply' field.
   - [x] Validate the 'URL' field appears.
7. Select 'Email' in the 'How to apply' field.
   - [x] Validate the 'Email' field appears.

### Test validations

As a content admin

8. Visit https://cms-fwlvssypud9glye0p1gfv4hmamkufgpi.ci.cms.va.gov/node/add/event
10. Select the 'Include registration information' checkbox
11. Select the 'Apply' value in the 'Call to action' field
12. Select the 'URL' option in the 'How to apply' field
13. Enter "mailto:test@example.com" in the 'URL' field.
14. Fill other required fields for the Event, and save.
   - [x] Validate you receive an error on the 'URL' field: "Links to email addresses are not allowed."
15. Change the 'How to apply' option to 'Email'
16. Enter "test@example.com" in the Email field, and save the node.
   - [x] Validate you receive the error message: "The email must be a VA.gov email address."

### Definition of Done


### Select Team for PR review
